Libplanet/Tx/Transaction.cs

```diff
@@ -225,62 +236,147 @@ public static Transaction<T> FromBencodex(byte[] bytes)
+        /// in order to fill the <see cref="UpdatedAddresses"/>.  See also
+        /// <em>Remarks</em> section.</param>
+        /// <param name="updatedAddresses"><see cref="Address"/>es whose
+        /// states affected by <paramref name="actions"/>.
+        /// These <see cref="Address"/>es are also included in
+        /// the <see cref="UpdatedAddresses"/> property, besides
+        /// <see cref="Address"/>es projected by evaluating
+        /// <paramref name="actions"/>.  See also <em>Remarks</em> section.
+        /// </param>
         /// <param name="timestamp">The time this <see cref="Transaction{T}"/>
         /// is created and signed.  This goes to the <see cref="Timestamp"/>
-        /// property.</param>
+        /// property.  If <c>null</c> (which is default) is passed this will
+        /// be the current time.</param>
         /// <returns>A created new <see cref="Transaction{T}"/> signed by
         /// the given <paramref name="privateKey"/>.</returns>
         /// <exception cref="ArgumentNullException">Thrown when <c>null</c>
         /// is passed to <paramref name="privateKey"/> or
         /// or <paramref name="actions"/>.
         /// </exception>
-        public static Transaction<T> Make(
+        public static Transaction<T> Create(
             PrivateKey privateKey,
-            Address recipient,
-            IList<T> actions,
-            DateTimeOffset timestamp)
+            IEnumerable<T> actions,
+            IImmutableSet<Address> updatedAddresses = null,
+            DateTimeOffset? timestamp = null
+        )
         {
-            if (privateKey == null)
+            if (ReferenceEquals(privateKey, null))
             {
                 throw new ArgumentNullException(nameof(privateKey));
             }
 
             PublicKey publicKey = privateKey.PublicKey;
-            var sender = new Address(publicKey);
+            var signer = new Address(publicKey);
+
+            if (ReferenceEquals(updatedAddresses, null))
+            {
+                updatedAddresses = ImmutableHashSet<Address>.Empty;
+            }
 
-            var tx = new Transaction<T>(
-                sender,
+            DateTimeOffset ts = timestamp ?? DateTimeOffset.UtcNow;
+
+            ImmutableArray<T> actionsArray = actions.ToImmutableArray();
+            byte[] payload = new Transaction<T>(
+                signer,
                 publicKey,
-                recipient,
-                timestamp,
-                actions
-            );
+                updatedAddresses,
+                ts,
+                actionsArray
+            ).ToBencodex(false);
+
+            if (!actionsArray.IsEmpty)
+            {
+                IAccountStateDelta delta = new Transaction<T>(
+                    signer,
+                    publicKey,
+                    updatedAddresses,
+                    ts,
+                    actionsArray
+                ).EvaluateActions(
```

Comment:
It's important to point out that `IAction.Execute()` is being executed in `Transaction<T>.Create()`.

Reply:
It's for filling out `UpdatedAddresses` automatically, and documented in `<remarks>` section.